File: src/http_types/FetchCacheMode.rs
/// https://developer.mozilla.org/en-US/docs/Web/API/Request/cache
#[repr(u8)]
#[derive(Copy, Clone, Eq, PartialEq, Debug, strum::IntoStaticStr)]
pub enum FetchCacheMode {
    #[strum(serialize = "default")]
    Default,
    #[strum(serialize = "no-store")]
    NoStore,
    #[strum(serialize = "reload")]
    Reload,
    #[strum(serialize = "no-cache")]
    NoCache,
    #[strum(serialize = "force-cache")]
    ForceCache,
    #[strum(serialize = "only-if-cached")]
    OnlyIfCached,
}

bun_core::comptime_string_map! {
    pub static MAP: FetchCacheMode = {
        b"default" => FetchCacheMode::Default,
        b"no-store" => FetchCacheMode::NoStore,
        b"reload" => FetchCacheMode::Reload,
        b"no-cache" => FetchCacheMode::NoCache,
        b"force-cache" => FetchCacheMode::ForceCache,
        b"only-if-cached" => FetchCacheMode::OnlyIfCached,
    };
}

impl FetchCacheMode {
    /// The map type is a zero-sized handle, so this is the same map as the
    /// module-level `MAP` static.
    pub const MAP: __ComptimeStringMap_MAP = __ComptimeStringMap_MAP(());
    // to_js lives as an extension-trait method in bun_http_jsc (see PORTING.md §Idiom map).
}

File: src/http_types/FetchCacheMode.zig
/// https://developer.mozilla.org/en-US/docs/Web/API/Request/cache
pub const FetchCacheMode = enum(u3) {
    default,
    @"no-store",
    reload,
    @"no-cache",
    @"force-cache",
    @"only-if-cached",

    pub const Map = bun.ComptimeStringMap(FetchCacheMode, .{
        .{ "default", .default },
        .{ "no-store", .@"no-store" },
        .{ "reload", .reload },
        .{ "no-cache", .@"no-cache" },
        .{ "force-cache", .@"force-cache" },
        .{ "only-if-cached", .@"only-if-cached" },
    });
    pub const toJS = @import("../http_jsc/fetch_enums_jsc.zig").fetchCacheModeToJS;
};

const bun = @import("bun");
